src/background/services/accounts/AccountsService.ts

```diff
@@ -82,10 +84,12 @@ export class AccountsService implements OnLock, OnUnlock {
 
   constructor(
     private storageService: StorageService,
-    private walletService: WalletService,
     private networkService: NetworkService,
     private permissionsService: PermissionsService,
-    private analyticsServicePosthog: AnalyticsServicePosthog
+    private analyticsServicePosthog: AnalyticsServicePosthog,
+    private secretsService: SecretsService,
+    private ledgerService: LedgerService,
+    private walletConnectService: WalletConnectService
```

Comment:
Should they be dependencies of the `SecretsService` instead? If I see correctly, there should be no circular dependencies and the only reason we inject them here is to pass them down to the `SecretsService` calls later on.

Reply:
The reason is why the dependencies got out of the SecretsService and took place in the AccountsService, because we talked about these cases with Gergo and the goal would be the SecretsService should have only one dependency in general, the StorageService, because it communicates basically only that service and the others are used in special cases in "its own" individual functions so the SecretService does not need those services in general. I know they are just moved to an other file in this case but it seems to be more rational.

Reply:
Makes sense 👍